### ISSUE TITLE
Crash on opening file on macOS

### ISSUE THREAD

**Issue by koolfreak** [Wavesonics/compose-multiplatform-file-picker#1]
When opening a file chooser sudden crash

Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'

osx 12.6.2, java 17.0.4.1

**Comment by Wavesonics** [comment]
Looks like it's throwing this exception:
`'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'`

Tried putting the function call on the Main dispatcher, but doesn't seem to fix it, still looking into it.

**Comment by shalva97** [comment]
Will you consider a PR to use jawa.awt.FileDialog? Using it works on my MacOS, have not checked on other platforms but I think it will also work there

**Comment by Wavesonics** [comment]
As a fallback that is fine, but the point of this library is to invoke the native platform dialog, so we really need to fix this problem.

I have very minimal experience programming for MacOS, so anyone who has more experience and could help on this issue would be greatly appreciated.

**Comment by shalva97** [comment]
> the point of this library is to invoke the native platform dialog

Yes, that is what `jawa.awt.FileDialog` does. Here is the pic

<img width="1093" alt="Screenshot 2023-05-06 at 21 56 07" src="https://user-images.githubusercontent.com/22417494/236639549-87f65eba-7731-4c0a-89e6-8a5abada17b2.png">

**Comment by shalva97** [comment]
What was the cases of `NativeFileDialog.NFD_PickFolder` and `JFileChooser` failing? In a way it is strange to add a third way of picking a file

**Comment by Wavesonics** [comment]
I could be wrong here, but I think that is just a well styled Java window, not the actual native picker dialog.

**Comment by shalva97** [comment]
Yes, [it is native](https://www.google.com/search?q=macos+ventura+native+file+dialog&tbm=isch&ved=2ahUKEwj40Jb9o-H-AhVfg_0HHV76BoAQ2-cCegQIABAA&oq=macos+ventura+native+file+dialog&gs_lcp=CgNpbWcQAzoECCMQJ1CmCFiCEGDiEWgAcAB4AIABkAKIAcoNkgEFMC42LjOYAQCgAQGqAQtnd3Mtd2l6LWltZ8ABAQ&sclient=img&ei=35ZWZLjlJd-G9u8P3vSbgAg&bih=790&biw=1390&client=firefox-b-d#imgrc=WxTRnq0U6RDPjM). Thats what opens for other applicatiions.

**Comment by Wavesonics** [comment]
`JFileChooser` is purely a swing window, so we dont want that.

`NativeFileDialog.NFD_PickFolder` is working except on MacOS and only due to a threading issue, so we just need to fix that and we should be good to go.

**Comment by Wavesonics** [comment]
I was looking for something in the docs that explicitly states that it is invoking the native dialog, but either way, I agree it sure looks like it is, so ya, I'd love a PR.

**Comment by shalva97** [comment]
yes, I forgot to comment on that. It seems to be library related.

So [here](https://github.com/LWJGL/lwjgl3/issues/697) they disscuss similar error. and when I read this comment https://github.com/LWJGL/lwjgl3/issues/697#issuecomment-971797157 I lost hopes.

Also tried to update lwjglVersion to `3.3.2` but it had same issue.

So yeah, I gave up on trying to fix `NativeFileDialog.NFD_OpenDialog`...

**Comment by shalva97** [comment]
> I was looking for something in the docs that explicitly states that it is invoking the native dialog

Also canot find it, but I think its by default that way, because AWT aims to use native components.

From Wikipedia:
> AWT widgets provided a thin level of abstraction over the underlying native user-interface. For example, creating an AWT [check box](https://en.wikipedia.org/wiki/Check_box) would cause AWT directly to call the underlying native subroutine that created a check box

**Comment by shalva97** [comment]
fuck, I cant believe there is no fully working native file chooser in Java. 

- `NativeFileDialog.NFD_OpenDialog` from lwjgl-> As it says here https://github.com/LWJGL/lwjgl3/issues/697#issuecomment-971797157 it does not work on MacOS
- AWT FileChooser -> cant pick directories and cant set proper filters (it does have `FilenameFilter`, but it works in a strange way)
- SWT -> cant even add the dependency this to the project, I know this is my fault but I... I give up after 1 hour of just trying to add a dependency
- JFileChooser -> is not native
- JavaFX -> Also cant add dependency: ` Failed to notify project evaluation listener.
    com.android.build.gradle.internal.BadPluginException: The 'java' plugin has been applied, but it is not compatible with the Android plugins.`

Maybe you can try with SWT? i just cant add the dependency :/ at this point I would go and write my own library to call it from JVM...

**Comment by overpas** [comment]
@Wavesonics @shalva97 Is there some workaround for this?

**Comment by Wavesonics** [comment]
so we have an idea, but it's going to take some time to implement

**Comment by HuixingWong** [comment]
Any update ?

**Comment by shalva97** [comment]
was working on a fully native solution here https://github.com/Wavesonics/compose-multiplatform-file-picker/pull/19, it felt cool, you know, its all Kotlin and stuff... 

The right solution is to fix existing code. So I looked at other projects, researched a bit more and managed to add dependency to TinyFileDialogs and it seems to work now https://github.com/Wavesonics/compose-multiplatform-file-picker/pull/24. will be happy for the feedback

**Comment by Wavesonics** [comment]
I tested it, looks good to me, closing
